### PR TITLE
Add API to set a custom user id

### PR DESCRIPTION
**Pull request by rodrilemos**
- Add support for SetCustomUserId (for analytics)
- Add support for SetUserType 
- Refactor Android JNI calls by a testable implementation
- Adde bunch of tests

### CODE REVIEW

File: Assets/ScaleMonk Ads/Source/UserType.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace ScaleMonk.Ads
+{
+    public enum UserType 
+    {
+        NON_PAYING_USER, PAYING_USER
```
Comment (arielf-camacho) [inline Assets/ScaleMonk Ads/Source/UserType.cs:5]: @marinoluck the convention in C# for Enum types is UpperCamelCase.

**Review by arielf-camacho — COMMENTED** [review]
LGTM!

File: Assets/ScaleMonk Ads/Source/UserType.cs (hunk 1 above)
Comment (marinoluck) [reply Assets/ScaleMonk Ads/Source/UserType.cs:5]: 👍

File: Assets/ScaleMonk Ads/Source/Android/AdsBinding.java [hunk 2]
```diff
@@ -3,32 +3,38 @@
...
 import com.scalemonk.ads.InterstitialEventListener;
 import com.scalemonk.ads.RewardedEventListener;
-import com.scalemonk.ads.BannerEventListener;
 import com.scalemonk.ads.ScaleMonkAds;
 import com.scalemonk.ads.unity.banner.BannerContainerFactory;
+import com.scalemonk.libs.ads.core.domain.UserType;
+import com.scalemonk.libs.ads.core.domain.session.UserTypeProvider;
 import com.unity3d.player.UnityPlayer;
 
-@Keep
```
Comment (nravasi) [inline Assets/ScaleMonk Ads/Source/Android/AdsBinding.java:16]: do not delete keep

**Review by nravasi — CHANGES_REQUESTED** [review]

**Review by asavizky-sm — APPROVED** [review]
LGTM! Just one nit

File: Assets/ScaleMonk Ads/Source/AdsEditorBinding.cs [hunk 3]
```diff
@@ -5,22 +5,21 @@
 // https://www.scalemonk.com/legal/en-US/mediation-license-agreement/index.html 
 //
 
-using System;
-using System.Collections.Generic;
-using System.Runtime.CompilerServices;
+using JetBrains.Annotations;
```
[the remark this answers is not in the document]
Comment (marinoluck) [reply Assets/ScaleMonk Ads/Source/AdsEditorBinding.cs:8]: I'm not sure, but we are also using it on all the java classes... I'll remove it It is used for NotNull check

[the remark this answers is not in the document]
Comment (marinoluck) [reply Assets/ScaleMonk Ads/Source/AdsEditorBinding.cs:8]: I'll keep it is used only for editor binding

**Review by nravasi — APPROVED** [review]

**Review by lsebrie — APPROVED** [review]

File: Assets/ScaleMonk Ads/Source/Android/AdsBinding.java [hunk 4]
```diff
@@ -59,24 +66,34 @@ public void showInterstitial(final Context context, final String tag) {
...
                 }
         );
     }
-    
+
     public void stopBanner(final Context context, final String tag) {
-        if (currentBannerContainer != null) {
-            this.activity.runOnUiThread(() -> ScaleMonkAds.stopBanner(context, currentBannerContainer, tag));
-        }
+        this.activity.runOnUiThread(() -> {
```
[the remark this answers is not in the document]
Comment (lsebrie) [reply Assets/ScaleMonk Ads/Source/Android/AdsBinding.java:89]: Also remember the iOS feature!

**Comment by marinoluck** [comment]
se cierra porque se movió todo acá https://github.com/scalemonk/mediation-sdk-unity/pull/50

**Pull request closed without merging**